models/users.js

```diff
@@ -397,37 +407,150 @@ const addForVerification = async (userId, discordId, profileImageUrl, discordIma
+    if (imageType === "both") {
+      await documentRef.update({ "discord.approved": true, "profile.approved": true });
+    } else {
+      const imageVerificationType = imageType === "discord" ? "discord.approved" : "profile.approved";
+      await documentRef.update({ [imageVerificationType]: true });
+    }
+
+    // IF BOTH IMAGES ARE VERIFIED THEN REMOVE THE ENTRY
+    const photoVerificationObject = (await documentRef.get()).data();
+    if (photoVerificationObject.discord.approved && photoVerificationObject.profile.approved) {
+      await documentRef.update({ status: photoVerificationRequestStatus.APPROVED });
+      await updateUserPicture(
+        { url: photoVerificationObject.profile.url, publicId: photoVerificationObject.profile.publicId },
+        userId
+      );
+    }
+
+    return "User image data verified successfully";
   } catch (err) {
     logger.error("Error while Removing Verification Entry", err);
     throw err;
   }
 };
 
 /**
- * Removes if user passed a valid image; ignores if no unverified record
+ * Returns the user's image verification requests (PENDING only) based on the userId
  * @param userId {String} - RDS user Id
  * @return {Promise<{Object}|{Object}>}
  * @throws {Error} - If error occurs while fetching user's image verification entry
  */
-const getUserImageForVerification = async (userId) => {
+const getUserPhotoVerificationRequests = async (userId) => {
   try {
-    const verificationImagesSnapshot = await photoVerificationModel.where("userId", "==", userId).get();
+    const verificationImagesSnapshotQuery = photoVerificationModel.where(
+      "status",
+      "==",
+      photoVerificationRequestStatus.PENDING
+    );
+
+    const user = await userModel.doc(userId).get();
+    if (!user.exists) {
+      throw new Error(`No document with userId: ${userId} was found!`);
+    }
+    const userData = user.data();
+    verificationImagesSnapshotQuery.where("userId", "==", userId);
+    const verificationImagesSnapshot = await verificationImagesSnapshotQuery.get();
     if (verificationImagesSnapshot.empty) {
       throw new Error(`No document with userId: ${userId} was found!`);
     }
-    return verificationImagesSnapshot.docs[0].data();
+
+    const photoVerificationObject = [
+      {
+        ...verificationImagesSnapshot.docs[0].data(),
+        user: {
+          username: userData.username,
+          picture: userData.picture.url,
+        },
+      },
+    ];
+    return photoVerificationObject;
+  } catch (err) {
+    logger.error("Error while Querying Photo Verification Entry", err);
+    throw err;
+  }
+};
+
+/**
+ * Returns all users image verification requests (PENDING only), optionally filtered by username
+ * @param username {String} - RDS username
+ * @return {Promise<{Object}|{Object}>}
+ * @throws {Error} - If error occurs while fetching user's image verification entry
+ */
+const getAllUsersPhotoVerificationRequests = async (username = null) => {
```

Comment:
Naming?

Reply:
is there any confusion or problem this may cause?